File: backend/src/main/java/com/inmobiliaria/persona/Persona.java
package com.inmobiliaria.persona;

import com.inmobiliaria.shared.multitenancy.TenantAware;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "personas")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Persona implements TenantAware {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "empresa_id", nullable = false)
    private Long empresaId;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_persona", nullable = false)
    private TipoPersona tipoPersona;

    // Persona Física
    private String nombre;

    @Column(name = "apellido_paterno")
    private String apellidoPaterno;

    @Column(name = "apellido_materno")
    private String apellidoMaterno;

    @Column(name = "fecha_nacimiento")
    private LocalDate fechaNacimiento;

    @Column(length = 18)
    private String curp;

    // Persona Moral
    @Column(name = "razon_social")
    private String razonSocial;

    @Column(name = "nombre_comercial")
    private String nombreComercial;

    // Común
    @Column(length = 13)
    private String rfc;

    private String email;

    private String telefono;

    @Column(name = "telefono_movil")
    private String telefonoMovil;

    @Column(nullable = false)
    private boolean activo = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "persona", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<PersonaRol> roles = new ArrayList<>();

    @OneToMany(mappedBy = "persona", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Direccion> direcciones = new ArrayList<>();

    @OneToMany(mappedBy = "persona", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<CuentaBancaria> cuentasBancarias = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public String getNombreCompleto() {
        if (tipoPersona == TipoPersona.MORAL) {
            return razonSocial;
        }
        StringBuilder sb = new StringBuilder();
        if (nombre != null) sb.append(nombre);
        if (apellidoPaterno != null) sb.append(" ").append(apellidoPaterno);
        if (apellidoMaterno != null) sb.append(" ").append(apellidoMaterno);
        return sb.toString().trim();
    }
}

File: frontend/src/types/persona.ts
export type TipoPersona = 'FISICA' | 'MORAL';
export type TipoDireccion = 'FISCAL' | 'CORRESPONDENCIA' | 'OTRA';

export interface Persona {
  id: number;
  empresaId: number;
  tipoPersona: TipoPersona;
  nombre?: string;
  apellidoPaterno?: string;
  apellidoMaterno?: string;
  fechaNacimiento?: string;
  curp?: string;
  razonSocial?: string;
  nombreComercial?: string;
  rfc?: string;
  email?: string;
  telefono?: string;
  telefonoMovil?: string;
  nombreCompleto: string;
  activo: boolean;
  createdAt?: string;
  updatedAt?: string;
  roles?: PersonaRol[];
  direcciones?: Direccion[];
  cuentasBancarias?: CuentaBancaria[];
}

export interface PersonaRol {
  id: number;
  rolId: number;
  rolNombre: string;
  fechaAsignacion: string;
  activo: boolean;
}

export interface Direccion {
  id: number;
  personaId: number;
  tipoDireccion: TipoDireccion;
  calle: string;
  numeroExterior?: string;
  numeroInterior?: string;
  estadoId?: number;
  estadoNombre?: string;
  municipioId?: number;
  municipioNombre?: string;
  coloniaId?: number;
  coloniaNombre?: string;
  codigoPostal?: string;
  referencias?: string;
  esPrincipal: boolean;
  activo: boolean;
}

export interface CuentaBancaria {
  id: number;
  personaId: number;
  banco: string;
  numeroCuenta?: string;
  clabe?: string;
  titular?: string;
  esPrincipal: boolean;
  activo: boolean;
}

export interface CreatePersonaRequest {
  tipoPersona: TipoPersona;
  nombre?: string;
  apellidoPaterno?: string;
  apellidoMaterno?: string;
  fechaNacimiento?: string;
  curp?: string;
  razonSocial?: string;
  nombreComercial?: string;
  rfc?: string;
  email?: string;
  telefono?: string;
  telefonoMovil?: string;
  rolesIds?: number[];
}

export interface UpdatePersonaRequest {
  tipoPersona?: TipoPersona;
  nombre?: string;
  apellidoPaterno?: string;
  apellidoMaterno?: string;
  fechaNacimiento?: string;
  curp?: string;
  razonSocial?: string;
  nombreComercial?: string;
  rfc?: string;
  email?: string;
  telefono?: string;
  telefonoMovil?: string;
  activo?: boolean;
}

export interface CreateDireccionRequest {
  tipoDireccion?: TipoDireccion;
  calle: string;
  numeroExterior?: string;
  numeroInterior?: string;
  estadoId?: number;
  municipioId?: number;
  coloniaId?: number;
  codigoPostal?: string;
  referencias?: string;
  esPrincipal?: boolean;
}

export interface CreateCuentaBancariaRequest {
  banco: string;
  numeroCuenta?: string;
  clabe?: string;
  titular?: string;
  esPrincipal?: boolean;
}
